app/__init__.py

```diff
@@ -114,13 +114,20 @@ def _get_field_filter(field_name, props):
     return field_type
 
 
+def _get_field_is_array(field_name, props):
+    return "array" == props.get("type")
+
+
 def process_spec(spec, process_unindexed=False):
     system_profile_spec_processed = {}
     for field, props in spec.items():
         if props.get("x-indexed", True) or process_unindexed:
-            field_type = _spec_type_to_python_type(props["type"])  # cast from string to type
             field_filter = _get_field_filter(field, props)
-            system_profile_spec_processed[field] = {"type": field_type, "filter": field_filter}
+            system_profile_spec_processed[field] = {
+                "type": _spec_type_to_python_type(props["type"]),  # cast from string to type
+                "filter": field_filter,
+                "is_array": _get_field_is_array(field, props),
```

Comment:
Since `_get_field_is_array` is only used here, should it just be inline?
```suggestion
                "is_array": "array" == props["type"],
```